lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHook.java

```diff
@@ -182,70 +183,69 @@ private NotificationEntity makeNotificationEntityWithIncludedFields(HookDoc hook
-            }
+            String valueString = postDoc.get(identityPath).asText();
+            entityData.add(new PathAndValue(pathString, valueString));
         }
-        List<String> changedPaths=new ArrayList<>();
-        List<NotificationEntity.PathAndValue> removedPaths=new ArrayList<>();
-        List<String> removedElements=new ArrayList<>();
-        for(DocComparator.Delta<JsonNode> delta:diff.getDelta()) {
-            if( (delta instanceof DocComparator.Move && arrayOrderSignificant) ||
-                !(delta instanceof DocComparator.Move) ) {
-
-                if(delta instanceof DocComparator.Removal) {
-                    JsonNode removedNode=((DocComparator.Removal<JsonNode>)delta).getRemovedNode();
-                    if(removedNode.isContainerNode()) {
-                        removedElements.add(delta.getField().toString());
-                        flatten(delta.getField().toString(),
-                                ((DocComparator.Removal<JsonNode>)delta).getRemovedNode(),
-                                removedPaths);
-                    } else {
-                        removedPaths.add(new NotificationEntity.PathAndValue(delta.getField().toString(),removedNode.asText()));
-                    }
+
+        // Add flattened include doc to entity data
+        JsonDoc includeDoc=includeProjector.project(postDoc,jsonNodeFactory);
+        flatten("", includeDoc.getRoot(), entityData);
+
+        // Add changes to entity data, removed entity data, changed paths, and removed elements
+        for(DocComparator.Delta<JsonNode> delta : diff.getDelta()) {
+            if (delta instanceof DocComparator.Move && arrayOrderSignificant) {
+                String newPath = delta.getField2().toString();
+                JsonNode movedNode = ((DocComparator.Move<JsonNode>) delta).getMovedNode();
+
+                changedPaths.add(newPath);
+                flatten(newPath, movedNode, entityData);
+            } else if (delta instanceof DocComparator.Removal) {
+                JsonNode removedNode=((DocComparator.Removal<JsonNode>)delta).getRemovedNode();
+                String removedPath = delta.getField().toString();
+
+                if(removedNode.isContainerNode()) {
+                    removedElements.add(removedPath);
+                    flatten(removedPath, removedNode, removedEntityData);
                 } else {
-                    if(delta instanceof DocComparator.Move) {
-                        // Add the new path to the changedPaths
-                        changedPaths.add(delta.getField2().toString());
-                    } else {
-                        changedPaths.add(delta.getField().toString());
-                    }
+                    String removedValue = removedNode.asText();
+                    removedEntityData.add(new PathAndValue(removedPath, removedValue));
                 }
+            } else if (delta instanceof DocComparator.Addition && hookDoc.getPreDoc() != null) {
```

Comment:
Yeah, sort of. Addition is, as I understand it, anything "new" that didn't exist in the json before. So like new array elements, or new object fields. On an insert, everything is new, so I made the to-be-revisited decision to not include data in those cases, because that would mean inserts get everything in the notification.